src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -150,25 +101,29 @@ private void checkOut(RoutingContext routingContext) {
       PatronActionSessionService.using(clients);
 
     ofAsync(() -> new LoanAndRelatedRecords(request.toLoan()))
-      .thenApply(servicePointOfCheckoutPresentValidator::refuseCheckOutWhenServicePointIsNotPresent)
-      .thenCombineAsync(userRepository.getUserByBarcode(request.getUserBarcode()), this::addUser)
-      .thenComposeAsync(r -> r.after(
-        automatedPatronBlocksValidator::refuseWhenCheckOutActionIsBlockedForPatron))
-      .thenCombineAsync(userRepository.getProxyUserByBarcode(request.getProxyUserBarcode()), this::addProxyUser)
-      .thenApply(inactiveUserValidator::refuseWhenUserIsInactive)
-      .thenApply(inactiveProxyUserValidator::refuseWhenUserIsInactive)
-      .thenCombineAsync(itemRepository.fetchByBarcode(request.getItemBarcode()), this::addItem)
-      .thenApply(itemNotFoundValidator::refuseWhenItemNotFound)
-      .thenApply(alreadyCheckedOutValidator::refuseWhenItemIsAlreadyCheckedOut)
-      .thenApply(itemStatusValidator::refuseWhenItemIsNotAllowedForCheckOut)
-      .thenComposeAsync(r -> r.after(proxyRelationshipValidator::refuseWhenInvalid))
-      .thenComposeAsync(r -> r.after(openLoanValidator::refuseWhenHasOpenLoan))
-      .thenComposeAsync(r -> r.after(requestQueueRepository::get))
-      .thenApply(requestedByAnotherPatronValidator::refuseWhenRequestedByAnotherPatron)
+      .thenApply(validators::refuseCheckOutWhenServicePointIsNotPresent)
+      .thenComposeAsync(r -> getUserByBarcode(request.getUserBarcode(), userRepository, errorHandler)
+        .thenApply(userResult -> addUser(r, userResult, errorHandler)))
+      .thenApply(validators::refuseWhenUserIsInactive)
```

Comment:
Right, previously it was:
- get **user**
- check that **user** is not blocked
- get **proxy user**
- check that **user** is active
- check that **proxy user** is active
I've moved the 4th check up, so that all user-related checks are grouped together, as well as proxy-user-related checks:
- get **user**
- check that **user** is active
- check that **user** is not blocked
- get **proxy user**
- check that **proxy user** is active
 This validation order makes a little more sense to me, but theoretically this may cause a failure with a different validation error in case when the user is both inactive and blocked. Apparently we don't have a test for this (otherwise the build would fail), probably because it wasn't important to anyone - check-out will be refused anyway. But I can reorder it back if you want to.

Reply:
If you think it's important to reorder it back to the previous order then please do it, but if you think that the theoretical validation error that may arise when the user is both inactive and blocked is gracefully handled then there is no need to restore the previous validation order.

Reply:
Actually, I've reordered it back. It's probably not important at all, but CIRC-1058 doesn't require this change and after all I think it's better to avoid these "while we're here" kind of improvements.